src/components/WearablePreview/EmoteControls.tsx

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable  @typescript-eslint/no-empty-function */
+import * as React from 'react'
+import { PreviewEmoteEventType } from '@dcl/schemas/dist/dapps/preview/preview-emote-event-type'
+import { IPreviewController } from '@dcl/schemas/dist/dapps/preview'
+import { Button } from '../Button/Button'
+import { Icon, WearablePreview } from '../..'
+import './EmoteControls.css'
+
+const ANIMATION_INTERVAL_PERCENTAGE = 0.1
+
+export type EmoteControlsProps = {
+  wearablePreviewId: string
+  className?: string
+  hideFrameInput?: boolean
+  hideProgressInput?: boolean
+  hidePlayButton?: boolean
+}
+
+type EmoteControlsState = {
+  frame: number
+  isPlaying: boolean
+  playingIntervalId?: number
+  previewController?: IPreviewController
+  length?: number
+}
+
+export class EmoteControls extends React.PureComponent<
+  EmoteControlsProps,
+  EmoteControlsState
+> {
+  state: EmoteControlsState = {
+    isPlaying: false,
+    frame: 0
+  }
+
+  async componentDidMount(): Promise<void> {
+    const previewController = WearablePreview.createController(
+      this.props.wearablePreviewId
+    )
+    const length = await previewController.emote.getLength()
```

Comment:
Since theres an `await` in this handler, everything that happens after it could happen after the component has been unmounted, and then the `setState` at the end of this handler would throw a warning/error. Usually when i need to do something async within a life cycle method I store on a class property a boolean telling if the component is mounted or not, like this:

```ts
class MyComponent extends Component {
  mounted = false
  componentDidMount() {
    mounted = true
  }
  componentWillUnmount() {
    mounted = false
  }
}
```

Then you can check after the await and return if the component was unmounted while the promise was resolving, like:

```ts
const length = await previewController.emote.getLength()
if (!this.mounted) return
// ...continue with the handler stuff
```

Reply:
excellent idea! updated to something similar to that :-)